flink-table-store-core/src/main/java/org/apache/flink/table/store/file/utils/RecordWriter.java

```diff
@@ -54,10 +54,10 @@
     /**
      * Prepare for a commit.
      *
-     * @param blocking if this method need to wait for current compaction to complete
+     * @param waitCompaction if this method need to wait for current compaction to complete
      * @return Incremental files in this snapshot cycle
      */
-    CommitIncrement prepareCommit(boolean blocking) throws Exception;
+    CommitIncrement prepareCommit(boolean waitCompaction) throws Exception;
```

Comment:
IMO, the `waitCompaction` has nothing to do with `prepareCommit`. On other words, the `compaction` operation should not be associated with the commit operation at the interface definition level.

Reply:
In the batch mode, we should wait compaction for the writer.